core/mondoo-ms365-security.mql.yaml

```diff
@@ -0,0 +1,625 @@
+        properties['passwordMinimumLength'] == 6
+      }
+      msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.macOSGeneralDeviceConfiguration") {
+        properties['passwordMinimumLength'] == 6
+      }
+      msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.iosGeneralDeviceConfiguration") {
+        properties['passcodeMinimumLength'] == 6
+      }
+      msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.androidGeneralDeviceConfiguration") {
+        properties['passwordMinimumLength'] == 6
+      }
+      msgraph.beta.devicemanagement.deviceConfigurations.where( properties["@odata.type"] == "#microsoft.graph.androidWorkProfileGeneralDeviceConfiguration") {
+        properties['passwordMinimumLength'] == 6
+      }
+  - uid: mondoo-m365-ensure-that-mobile-device-encryption-is-enabled-to-prevent-unauthorized-access-to-mobile-data
+    title: Ensure that Android mobile device encryption is enabled
+    severity: 80
+    docs:
+      desc: |
+        This control checks that encryption in android mobile devices has been enabled to prevent any unauthorized access to the data 
+      audit: |
+        __cnquery run__
+
+        To audit Microsoft MS365 with `cnquery run`:
+
+        Run the following query:
+
+          ```mql
+          cnquery run ms365 -c "msgraph.beta.devicemanagement.deviceConfigurations.where( properties['@odata.type'] == "#microsoft.graph.androidGeneralDeviceConfiguration") { properties['storageRequireDeviceEncryption'] == true }" --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        __cnquery shell__
+
+        To audit Microsoft ms365 with `cnquery shell`:
+
+        1. Launch `cnquery shell`:
+
+          ```bash
+          cnquery shell ms365 --certificate-path <*.pem> --tenant-id <tenant_id> --client-id <client_id>
+          ```
+        
+        2. Run the following query:
+
+          ```mql
+          msgraph.beta.devicemanagement.deviceConfigurations.where( properties['@odata.type'] == "#microsoft.graph.androidGeneralDeviceConfiguration") { properties['storageRequireDeviceEncryption'] == true }
+          ```
+      remediation: |
+        ### MS365 Console
+         
+         To update via the ms365 portal:
+
+         1. Log in as to the ms365 portal at https://admin.microsoft.com
+         2. Endpoint Manager --> Devices --> Policy --> Configuration profiles 
+         3. Ensure that a profile exists for Android with following conditions:
+            * Password section --> Device restrictions --> Encryption is set to require
+    query: |
+      msgraph.beta.devicemanagement.deviceConfigurations.where( properties['@odata.type'] == "#microsoft.graph.androidGeneralDeviceConfiguration") {
```

Comment:
`msgraph.beta` is deprecated, use latest MQL resources

Reply:
done!